### ISSUE TITLE
Booting is absurdly slow on 6.10.3-surface-1

### ISSUE THREAD

**Issue by ishland** [linux-surface/linux-surface#1525]
[Description of the bug or feature]
As the title says. 

### Environment

- Hardware model: Surface Pro 5
- Kernel version: `Linux version 6.10.3-surface-1 (root@773adc913830) (gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1 SMP PREEMPT_DYNAMIC Sun Aug  4 16:23:07 UTC 2024`
- Distribution: Debian testing

`dmesg` output for `6.10.3-surface-1` (outdated, see [this](https://github.com/linux-surface/linux-surface/issues/1525#issuecomment-2290961510)): https://gist.github.com/ishland/74f6a92bda4ae6c807faa680d69f4d25

`dmesg` output for `6.9.3-surface-2` (normal booting): https://gist.github.com/ishland/189ba883b47a85fddcfa7aa5d034b3fc

**Comment by tormodvolden** [comment]
Your 6.10.3 dmesg output contains:
```
[    0.111859] GDS: Microcode update needed! Disabling AVX as mitigation.
[    0.111863] GDS: Mitigation: AVX disabled, no microcode
```
I guess you also see a lot of "invalid opcode" traps when you run dmesg after booting up?

**Comment by tormodvolden** [comment]
`[    0.112666] smpboot: CPU0: Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz (family: 0x6, model: 0x8e, stepping: 0x9)`
You have a 06-8e-09 model, which receives an updated microcode in intel-microcode 20240813.

**Comment by ishland** [comment]
The microcode is simply missing in the UKI for 6.10.3 for some reasons and I've fixed it. 
GDS mitigation now works as expected, but the slow booting persists. 
New `dmesg` for `6.10.3-surface-1`: https://gist.github.com/ishland/3e580b177cc907963c7b06001865457c

I have also noticed that the 6.10 kernel that Debian ships have the same issue:
`dmesg` for `6.10.3-1`: https://gist.github.com/ishland/e410059549c16036f31768755efa0f56

**Comment by tormodvolden** [comment]
So you are not seeing "traps", "invalid opcode" or crashes (other than the wifi driver on the Debian kernel)?

**Comment by ishland** [comment]
> So you are not seeing "traps", "invalid opcode" or crashes (other than the wifi driver on the Debian kernel)?

Yeah I don't see any of these on the surface kernel or the Debian kernel.

**Comment by qzed** [comment]
There are a couple of lines like the one below for the surface kernel. I can't spot it in the Debian kernel log though. Still, maybe an indication as to where things are going wrong.

```
[    6.212797] workqueue: drm_fb_helper_damage_work hogged CPU for >10000us 35 times, consider switching to WQ_UNBOUND
```

**Comment by ishland** [comment]
`systemd-analyze plot`
[6.9.3-surface-2](https://github.com/user-attachments/assets/9798ee2a-3d00-4518-93e0-0d5fdfa3070a)
[6.10.3-surface-1](https://github.com/user-attachments/assets/1be889a0-f0c2-4dd1-ad88-11acaa6796c9)
[6.10.3-amd64 (debian kernel))](https://github.com/user-attachments/assets/3a1d014e-5207-4c14-8c28-092fa52df87d)

**Comment by tormodvolden** [comment]
I'd say the big difference is the ufw.service that takes 1s, 18, or 28s.

**Comment by qzed** [comment]
That's not the only difference. Some of the device services (e.g. `sys-module-configfs.device`) take almost twice as long... no idea why though.

**Comment by niiiksh** [comment]
Same on Surface Pro 4. Also after boot the login screen scaled wrong with 100% rate that must be 200% and then the screen goes black and can’t show anything. Sometimes it may “fix” scaling and the screen doesn’t freeze to black.

**Comment by ishland** [comment]
It is still very much a problem on linux 6.11. I tried the `6.11.2-amd64` kernel shipped by Debian and is still very slow.

[plot.svg](https://github.com/user-attachments/assets/0a204e80-53b7-473b-ad66-c446b7acfecc)

**Comment by ishland** [comment]
Seems to be fixed in 6.12.3:
![plot](https://github.com/user-attachments/assets/12bbd7c2-e0f8-48ae-ab49-cc3c5d8c5b19)
